src/main/app/forms/validation/validators/isLessThanOrEqualToSumOf.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+
+@ValidatorConstraint()
+export class IsLessThanOrEqualToSumOfConstraint implements ValidatorConstraintInterface {
+  validate (value: any, args?: ValidationArguments) {
+    if (value === undefined) {
+      return true
+    }
+    const argsLength = args.constraints.length
+    const expectedNumberOfArgs = 2
+    if (argsLength !== expectedNumberOfArgs) {
+      throw new Error(`Invalid number of arguments, got ${argsLength}, expected ${expectedNumberOfArgs}`)
+    }
+
+    const amountToSumProperty = args.constraints[0]
+    const lessThanAmountPropertyName = args.constraints[1]
+
+    const lessThanAmount = (args.object as any)[lessThanAmountPropertyName]
+    const amountToSum = (args.object as any)[amountToSumProperty]
+
+    // Not this validator's concern if the amount hasn't been entered
+    if (!amountToSum) {
+      return true
+    }
+
+    return typeof value === 'number'
+      && typeof lessThanAmount === 'number'
+      && typeof amountToSum === 'number'
+      && (value + amountToSum) <= lessThanAmount
+  }
+}
+
+export function IsLessThanOrEqualToSumOf (sum: string, lessThanAmount: string, validationOptions?: ValidationOptions) {
```

Comment:
I would say it would be nice to have some JSDoc comment saying what this validator is doing / what problem it is supposed to solve since it is quite complex one. Validator name does not fully uncover its complexity.

Reply:
I know, couldn't think of a better name :(